### PR TITLE
Fix tablet restore api in BE(#1623)

### PR DESCRIPTION
**Pull request by kangpinghuang**
1. fix bug when tablet with pending dir
2. add a shell script to support batch restore

### CODE REVIEW

File: be/src/http/action/restore_tablet_action.cpp [hunk 1]
```diff
@@ -86,11 +86,11 @@ Status RestoreTabletAction::_handle(HttpRequest *req) {
 
     TabletSharedPtr tablet =
             StorageEngine::instance()->tablet_manager()->get_tablet(tablet_id, schema_hash);
-    if (tablet.get() != nullptr) {
+    if (tablet = nullptr) {
```
Comment (imay) [inline be/src/http/action/restore_tablet_action.cpp]:
```suggestion
    if (tablet == nullptr) {
```

File: be/src/http/action/restore_tablet_action.cpp [hunk 2]
```diff
@@ -121,36 +121,27 @@ Status RestoreTabletAction::_reload_tablet(
                      << ", signature: " << tablet_id;
         // remove tablet data path in data path
         // path: /roo_path/data/shard/tablet_id
-        std::string tablet_path = shard_path + "/" + std::to_string(tablet_id);
+        std::string tablet_path = shard_path + "/" + std::to_string(tablet_id) + "/" + std::to_string(schema_hash);
```
Comment (imay) [inline be/src/http/action/restore_tablet_action.cpp]: You can use strings::Substitute to finish it

File: be/src/http/action/restore_tablet_action.cpp [hunk 3]
```diff
@@ -185,31 +176,48 @@ Status RestoreTabletAction::_restore(const std::string& key, int64_t tablet_id,
...
-            Status s = FileUtils::remove_all(restore_tablet_path);
+        std::string from = src + "/" + file;
+        std::string to = dst + "/" + file;
+        if (FileUtils::is_dir(from)) {
+            s = FileUtils::create_dir(to);
+            if (!s.ok()) {
+                LOG(WARNING) << "create path failed:" << to;
+                return s;
+            }
+            s = _create_hard_link_recursive(from, to);
```
Comment (imay) [inline be/src/http/action/restore_tablet_action.cpp]: RETURN_IF_ERROR()

File: be/src/http/action/restore_tablet_action.cpp [hunk 4]
```diff
@@ -185,31 +176,48 @@ Status RestoreTabletAction::_restore(const std::string& key, int64_t tablet_id,
...
+    }
+    std::string restore_shard_path = store->get_absolute_shard_path(std::to_string(tablet_meta.shard_id()));
+    Status status = _reload_tablet(key, restore_shard_path, tablet_id, schema_hash);
+    return status;
+}
+
+Status RestoreTabletAction::_create_hard_link_recursive(const std::string& src, const std::string& dst) {
     std::vector<std::string> files;
-    s = FileUtils::scan_dir(latest_tablet_path, &files);
+    Status s = FileUtils::scan_dir(src, &files);
```
Comment (imay) [inline be/src/http/action/restore_tablet_action.cpp]: RETURN_IF_ERROR

File: bin/restore_tablet_tool.sh [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# print usage
+usage() {
+    echo "
+Usage: $0 <options>
+    Optional options:
+        -h | --host  BE hostname
```
Comment (imay) [inline bin/restore_tablet_tool.sh]: usually -h means help. 
we can use -s to means server like "-s http://localhost:9070"

File: bin/restore_tablet_tool.sh [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# print usage
+usage() {
+    echo "
```
Comment (imay) [inline bin/restore_tablet_tool.sh]: give a example to let user copy

And you should give some comments for this command

**Review by imay — CHANGES_REQUESTED** [review]

File: be/src/http/action/restore_tablet_action.cpp (hunk 2 above)
Comment (kangpinghuang) [reply be/src/http/action/restore_tablet_action.cpp]: ok

File: be/src/http/action/restore_tablet_action.cpp (hunk 1 above)
Comment (kangpinghuang) [reply be/src/http/action/restore_tablet_action.cpp]: ok

File: be/src/http/action/restore_tablet_action.cpp (hunk 4 above)
Comment (kangpinghuang) [reply be/src/http/action/restore_tablet_action.cpp]: ok

File: be/src/http/action/restore_tablet_action.cpp (hunk 3 above)
Comment (kangpinghuang) [reply be/src/http/action/restore_tablet_action.cpp]: ok

File: bin/restore_tablet_tool.sh (hunk 5 above)
Comment (kangpinghuang) [reply bin/restore_tablet_tool.sh]: ok

File: bin/restore_tablet_tool.sh (hunk 6 above)
Comment (kangpinghuang) [reply bin/restore_tablet_tool.sh]: ok

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**